File: app/src/main/java/com/blankj/androidutilcode/feature/core/path/PathActivity.java
package com.blankj.androidutilcode.feature.core.path;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.View;
import android.widget.TextView;

import com.blankj.androidutilcode.R;
import com.blankj.androidutilcode.base.BaseBackActivity;
import com.blankj.utilcode.util.PathUtils;
import com.blankj.utilcode.util.SpanUtils;

/**
 * <pre>
 *     author: Blankj
 *     blog  : http://blankj.com
 *     time  : 2016/10/13
 *     desc  : demo about PathUtils
 * </pre>
 */
public class PathActivity extends BaseBackActivity {

    public static void start(Context context) {
        Intent starter = new Intent(context, PathActivity.class);
        context.startActivity(starter);
    }

    @Override
    public void initData(@Nullable Bundle bundle) {

    }

    @Override
    public int bindLayout() {
        return R.layout.activity_path;
    }

    @Override
    public void initView(Bundle savedInstanceState, View contentView) {
        getToolBar().setTitle(getString(R.string.demo_path));

        TextView tvAboutMetaData = findViewById(R.id.tv_about_path);
        tvAboutMetaData.setText(new SpanUtils()
                .appendLine("getRootPath: " + PathUtils.getRootPath())
                .appendLine("getDataPath: " + PathUtils.getDataPath())
                .appendLine("getDownloadCachePath: " + PathUtils.getDownloadCachePath())

                .appendLine("getInternalAppDataPath: " + PathUtils.getInternalAppDataPath())
                .appendLine("getInternalAppCodeCacheDir: " + PathUtils.getInternalAppCodeCacheDir())
                .appendLine("getInternalAppCachePath: " + PathUtils.getInternalAppCachePath())
                .appendLine("getInternalAppDbsPath: " + PathUtils.getInternalAppDbsPath())
                .appendLine("getInternalAppDbPath: " + PathUtils.getInternalAppDbPath("demo"))
                .appendLine("getInternalAppFilesPath: " + PathUtils.getInternalAppFilesPath())
                .appendLine("getInternalAppSpPath: " + PathUtils.getInternalAppSpPath())
                .appendLine("getInternalAppNoBackupFilesPath: " + PathUtils.getInternalAppNoBackupFilesPath())

                .appendLine("getExternalStoragePath: " + PathUtils.getExternalStoragePath())
                .appendLine("getExternalMusicPath: " + PathUtils.getExternalMusicPath())
                .appendLine("getExternalPodcastsPath: " + PathUtils.getExternalPodcastsPath())
                .appendLine("getExternalRingtonesPath: " + PathUtils.getExternalRingtonesPath())
                .appendLine("getExternalAlarmsPath: " + PathUtils.getExternalAlarmsPath())
                .appendLine("getExternalNotificationsPath: " + PathUtils.getExternalNotificationsPath())
                .appendLine("getExternalPicturesPath: " + PathUtils.getExternalPicturesPath())
                .appendLine("getExternalMoviesPath: " + PathUtils.getExternalMoviesPath())
                .appendLine("getExternalDownloadsPath: " + PathUtils.getExternalDownloadsPath())
                .appendLine("getExternalDcimPath: " + PathUtils.getExternalDcimPath())
                .appendLine("getExternalDocumentsPath: " + PathUtils.getExternalDocumentsPath())

                .appendLine("getExternalAppDataPath: " + PathUtils.getExternalAppDataPath())
                .appendLine("getExternalAppCachePath: " + PathUtils.getExternalAppCachePath())
                .appendLine("getExternalAppFilesPath: " + PathUtils.getExternalAppFilesPath())
                .appendLine("getExternalAppMusicPath: " + PathUtils.getExternalAppMusicPath())
                .appendLine("getExternalAppPodcastsPath: " + PathUtils.getExternalAppPodcastsPath())
                .appendLine("getExternalAppRingtonesPath: " + PathUtils.getExternalAppRingtonesPath())
                .appendLine("getExternalAppAlarmsPath: " + PathUtils.getExternalAppAlarmsPath())
                .appendLine("getExternalAppNotificationsPath: " + PathUtils.getExternalAppNotificationsPath())
                .appendLine("getExternalAppPicturesPath: " + PathUtils.getExternalAppPicturesPath())
                .appendLine("getExternalAppMoviesPath: " + PathUtils.getExternalAppMoviesPath())
                .appendLine("getExternalAppDownloadPath: " + PathUtils.getExternalAppDownloadPath())
                .appendLine("getExternalAppDcimPath: " + PathUtils.getExternalAppDcimPath())
                .appendLine("getExternalAppDocumentsPath: " + PathUtils.getExternalAppDocumentsPath())
                .appendLine("getExternalAppObbPath: " + PathUtils.getExternalAppObbPath())
                .append("")
                .create());
    }

    @Override
    public void doBusiness() {

    }

    @Override
    public void onWidgetClick(View view) {

    }
}

File: feature/utilcode/pkg/src/main/java/com/blankj/utilcode/pkg/feature/path/PathActivity.kt
package com.blankj.utilcode.pkg.feature.path

import android.content.Context
import android.content.Intent
import com.blankj.common.activity.CommonActivity
import com.blankj.common.item.CommonItem
import com.blankj.common.item.CommonItemTitle
import com.blankj.utilcode.pkg.R
import com.blankj.utilcode.util.CollectionUtils
import com.blankj.utilcode.util.PathUtils

/**
 * ```
 * author: Blankj
 * blog  : http://blankj.com
 * time  : 2016/10/13
 * desc  : demo about PathUtils
 * ```
 */
class PathActivity : CommonActivity() {

    companion object {
        fun start(context: Context) {
            val starter = Intent(context, PathActivity::class.java)
            context.startActivity(starter)
        }
    }

    override fun bindTitleRes(): Int {
        return R.string.demo_path
    }

    override fun bindItems(): MutableList<CommonItem<*>> {
        return CollectionUtils.newArrayList(
                CommonItemTitle("getRootPath", PathUtils.getRootPath()),
                CommonItemTitle("getDataPath", PathUtils.getDataPath()),
                CommonItemTitle("getDownloadCachePath", PathUtils.getDownloadCachePath()),

                CommonItemTitle("getInternalAppDataPath", PathUtils.getInternalAppDataPath()),
                CommonItemTitle("getInternalAppCodeCacheDir", PathUtils.getInternalAppCodeCacheDir()),
                CommonItemTitle("getInternalAppCachePath", PathUtils.getInternalAppCachePath()),
                CommonItemTitle("getInternalAppDbsPath", PathUtils.getInternalAppDbsPath()),
                CommonItemTitle("getInternalAppDbPath", PathUtils.getInternalAppDbPath("demo")),
                CommonItemTitle("getInternalAppFilesPath", PathUtils.getInternalAppFilesPath()),
                CommonItemTitle("getInternalAppSpPath", PathUtils.getInternalAppSpPath()),
                CommonItemTitle("getInternalAppNoBackupFilesPath", PathUtils.getInternalAppNoBackupFilesPath()),

                CommonItemTitle("getExternalStoragePath", PathUtils.getExternalStoragePath()),
                CommonItemTitle("getExternalMusicPath", PathUtils.getExternalMusicPath()),
                CommonItemTitle("getExternalPodcastsPath", PathUtils.getExternalPodcastsPath()),
                CommonItemTitle("getExternalRingtonesPath", PathUtils.getExternalRingtonesPath()),
                CommonItemTitle("getExternalAlarmsPath", PathUtils.getExternalAlarmsPath()),
                CommonItemTitle("getExternalNotificationsPath", PathUtils.getExternalNotificationsPath()),
                CommonItemTitle("getExternalPicturesPath", PathUtils.getExternalPicturesPath()),
                CommonItemTitle("getExternalMoviesPath", PathUtils.getExternalMoviesPath()),
                CommonItemTitle("getExternalDownloadsPath", PathUtils.getExternalDownloadsPath()),
                CommonItemTitle("getExternalDcimPath", PathUtils.getExternalDcimPath()),
                CommonItemTitle("getExternalDocumentsPath", PathUtils.getExternalDocumentsPath()),

                CommonItemTitle("getExternalAppDataPath", PathUtils.getExternalAppDataPath()),
                CommonItemTitle("getExternalAppCachePath", PathUtils.getExternalAppCachePath()),
                CommonItemTitle("getExternalAppFilesPath", PathUtils.getExternalAppFilesPath()),
                CommonItemTitle("getExternalAppMusicPath", PathUtils.getExternalAppMusicPath()),
                CommonItemTitle("getExternalAppPodcastsPath", PathUtils.getExternalAppPodcastsPath()),
                CommonItemTitle("getExternalAppRingtonesPath", PathUtils.getExternalAppRingtonesPath()),
                CommonItemTitle("getExternalAppAlarmsPath", PathUtils.getExternalAppAlarmsPath()),
                CommonItemTitle("getExternalAppNotificationsPath", PathUtils.getExternalAppNotificationsPath()),
                CommonItemTitle("getExternalAppPicturesPath", PathUtils.getExternalAppPicturesPath()),
                CommonItemTitle("getExternalAppMoviesPath", PathUtils.getExternalAppMoviesPath()),
                CommonItemTitle("getExternalAppDownloadPath", PathUtils.getExternalAppDownloadPath()),
                CommonItemTitle("getExternalAppDcimPath", PathUtils.getExternalAppDcimPath()),
                CommonItemTitle("getExternalAppDocumentsPath", PathUtils.getExternalAppDocumentsPath()),
                CommonItemTitle("getExternalAppObbPath", PathUtils.getExternalAppObbPath())
        )
    }
}
